core/src/main/java/com/oath/oak/OakDetachedReadBuffer.java

```diff
@@ -91,7 +91,7 @@ public double getDouble(int index) {
         if (transformer == null) {
             throw new NullPointerException();
         }
-        return safeAccessToAttachedBuffer(buffer -> transformer.apply(buffer.getDuplicatedReadByteBuffer().slice()));
+        return safeAccessToAttachedBuffer(transformer::apply);
```

Comment:
I wonder if this syntactic sugar also implies some performance degradation...

Reply:
I believe that the cost of using lambda come from the need to instantiate a new object.
The `transformer::apply` "syntactic sugar" is actually a necessary syntax to allow usage of exiting class methods as input to functions that accept a lambda function, without having to instantiate a new object (i.e., the lambda function object).
So we actually improve performance when we use this syntax, which implies it is not just syntactic sugar.

The actual possible performance issue is a few lines above in the `get{Something}()` methods that instantiate a new lambda function for each get operation.
But we currently do not have any benchmark that evaluates the detached buffers.

Reply:
Agree. Just add a TODO comment for
> The actual possible performance issue is a few lines above in the get{Something}() methods that instantiate a new lambda function for each get operation.

Reply:
Found a workaround that avoids instantiating new lambda objects.